src/builder/Actions.sol

```diff
@@ -681,48 +612,56 @@ library Actions {
         return (quarkOperation, action);
     }
 
-    function unwrapAsset(UnwrapAsset memory unwrap, PaymentInfo.Payment memory payment, bool useQuotecall)
-        internal
-        pure
-        returns (IQuarkWallet.QuarkOperation memory, Action memory)
-    {
+    function transformToConterpartAsset(
+        WrapOrUnwrapAsset memory wrapOrUnwrap,
+        PaymentInfo.Payment memory payment,
+        bool useQuotecall
+    ) internal pure returns (IQuarkWallet.QuarkOperation memory, Action memory) {
         bytes[] memory scriptSources = new bytes[](1);
         scriptSources[0] = type(WrapperActions).creationCode;
 
-        Accounts.ChainAccounts memory accounts = Accounts.findChainAccounts(unwrap.chainId, unwrap.chainAccountsList);
+        Accounts.ChainAccounts memory accounts =
+            Accounts.findChainAccounts(wrapOrUnwrap.chainId, wrapOrUnwrap.chainAccountsList);
 
         Accounts.AssetPositions memory assetPositions =
-            Accounts.findAssetPositions(unwrap.assetSymbol, accounts.assetPositionsList);
-
-        Accounts.QuarkState memory accountState = Accounts.findQuarkState(unwrap.sender, accounts.quarkStates);
+            Accounts.findAssetPositions(wrapOrUnwrap.assetSymbol, accounts.assetPositionsList);
 
+        Accounts.QuarkState memory accountState = Accounts.findQuarkState(wrapOrUnwrap.sender, accounts.quarkStates);
         // Construct QuarkOperation
         IQuarkWallet.QuarkOperation memory quarkOperation = IQuarkWallet.QuarkOperation({
             nonce: accountState.quarkNextNonce,
             scriptAddress: CodeJarHelper.getCodeAddress(type(WrapperActions).creationCode),
-            scriptCalldata: TokenWrapper.encodeUnwrapToken(unwrap.chainId, unwrap.assetSymbol, unwrap.amount),
+            scriptCalldata: TokenWrapper.encodeActionToTransformToCounterpart(
```

Comment:
Oops sorry to go back and forth, but now that you renamed action context to `WrapOrUnwrap`, it seems fine to just name this `encodeActionToWrapOrUnwrap`. `encodeActionToTransformToCounterpart` is a bit hard to understand